core/templates/dev/head/domain/story/StoryObjectFactory.js

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating and mutating instances of frontend
+ * story domain objects.
+ */
+
+oppia.factory('StoryObjectFactory', ['StoryContentsObjectFactory',
+  function(StoryContentsObjectFactory) {
+    var Story = function(
+        id, title, description, notes, storyContents, languageCode, version) {
+      this._id = id;
+      this._title = title;
+      this._description = description;
+      this._notes = notes;
+      this._storyContents = storyContents;
+      this._languageCode = languageCode;
+      this._version = version;
+    };
+
+    // Instance methods
+
+    Story.prototype.getId = function() {
+      return this._id;
+    };
+
+    Story.prototype.getTitle = function() {
+      return this._title;
+    };
+
+    Story.prototype.setTitle = function(title) {
+      this._title = title;
+    };
+
+    Story.prototype.getDescription = function() {
+      return this._description;
+    };
+
+    Story.prototype.setDescription = function(description) {
+      this._description = description;
+    };
+
+    Story.prototype.getNotes = function() {
+      return this._notes;
+    };
+
+    Story.prototype.setNotes = function(notes) {
+      this._notes = notes;
+    };
+
+    Story.prototype.getLanguageCode = function() {
+      return this._languageCode;
+    };
+
+    Story.prototype.setLanguageCode = function(languageCode) {
+      this._languageCode = languageCode;
+    };
+
+    Story.prototype.getVersion = function() {
+      return this._version;
+    };
+
+    Story.prototype.getStoryContents = function() {
+      return this._storyContents;
+    };
+
+    // Reassigns all values within this story to match the existing
+    // story. This is performed as a deep copy such that none of the
+    // internal, bindable objects are changed within this story.
+    Story.prototype.copyFromStory = function(otherStory) {
```

Comment:
In what case would you need this method?

Reply:
In StoryEditorStateService, this would be used to set the story after creating an interstitial story.